feat(formatter): keep parser options consistent for all formatter usages

Dunqing opened the pull request:
Copy from oxfmt
https://github.com/oxc-project/oxc/blob/adda5f56de542ce178f0c68f5732d60166c278b2/apps/oxfmt/src/service.rs#L45-L54

Dunqing commented:
* **#13885** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13885?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#13884** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13884?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13884?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=13884).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

leaysgur reviewed (approved):
We will add `Formatter::parser_options()` one day?

Dunqing committed:
feat(formatter): keep parser options consistent for all formatter usages (#13884)

Copy from oxfmt
https://github.com/oxc-project/oxc/blob/adda5f56de542ce178f0c68f5732d60166c278b2/apps/oxfmt/src/service.rs#L45-L54

Dunqing commented:
> We will add `Formatter::parser_options()` one day?

Yes, It should be.
